lib/inertia_rails/renderer.rb

```diff
@@ -27,13 +27,13 @@ def render
     private
 
     def props
-      only = (@request.headers['X-Inertia-Partial-Data'] || '').split(',').compact.map(&:to_sym)
-
-      _props = ::InertiaRails.shared_data(@controller).merge(@props)
-
-      _props = (only.any? && @request.headers['X-Inertia-Partial-Component'] == component) ?
-        _props.select {|key| key.in? only} :
-        _props
+      _props = ::InertiaRails.shared_data(@controller).merge(@props).select do |key, prop|
+        if rendering_partial_component?
+          key.in? partial_keys
+        else
+          !prop.is_a?(InertiaRails::Lazy)
```

Comment:
This sets off my duck-typing alarm bells a bit, but off the top of my head, I can't think of a better option since the alternative is various "primitives"...

Reply:
I could definitely see a future where _all_ props become some kind of polymorphic prop class so this logic gets pulled out of the renderer and there is a more consistent treatment of the "kinds" of props (plain data, blocks, and lazy)